src/cargo/sources/registry/http_remote.rs

```diff
@@ -111,8 +124,8 @@ struct Download {
     /// Actual downloaded data, updated throughout the lifetime of this download.
     data: RefCell<Vec<u8>>,
 
-    /// ETag or Last-Modified header received from the server (if any).
-    index_version: RefCell<Option<String>>,
+    /// HTTP headers.
+    header_map: RefCell<HashMap<String, Vec<String>>>,
```

Comment:
Should we have a HashMap here, or list out the headers we actually care about? Either a variable per header we care about, or a data structure with each of them as separate fields.